packages/mdc-toolbar/component.ts

```diff
@@ -23,38 +23,28 @@
 
 import {MDCComponent} from '@material/base/component';
 import {MDCRipple} from '@material/ripple/index';
+import {MDCToolbarFoundation} from './foundation';
+import {ToolbarEventDetail} from './types';
 
-import MDCToolbarFoundation from './foundation';
+const strings = MDCToolbarFoundation.strings;
 
-export {MDCToolbarFoundation};
-
-export class MDCToolbar extends MDCComponent {
-  static attachTo(root) {
+export class MDCToolbar extends MDCComponent<MDCToolbarFoundation> {
+  static attachTo(root: Element) {
     return new MDCToolbar(root);
   }
 
-  get firstRowElement_() {
-    return this.root_.querySelector(MDCToolbarFoundation.strings.FIRST_ROW_SELECTOR);
-  }
-
-  get titleElement_() {
-    return this.root_.querySelector(MDCToolbarFoundation.strings.TITLE_SELECTOR);
-  }
-
-  set fixedAdjustElement(fixedAdjustElement) {
-    this.fixedAdjustElement_ = fixedAdjustElement;
-    this.foundation_.updateAdjustElementStyles();
-  }
+  protected root_!: HTMLElement; // assigned in MDCComponent constructor
 
-  get fixedAdjustElement() {
-    return this.fixedAdjustElement_;
-  }
+  private ripples_!: MDCRipple[]; // assigned in initialize()
+  private fixedAdjustElement_?: HTMLElement;
 
   initialize() {
-    this.ripples_ = [].map.call(this.root_.querySelectorAll(MDCToolbarFoundation.strings.ICON_SELECTOR), (icon) => {
+    this.ripples_ = [];
+
+    [].forEach.call(this.root_.querySelectorAll(strings.ICON_SELECTOR), (icon: HTMLElement) => {
```

Comment:
Did the existing `[].map` not work?

Reply:
It technically worked, but it was more cumbersome and less type safe. TS can't infer the return type of the array, so I had to cast it to `MDCRipple[]`:

```ts
this.ripples_ = [].map.call(this.root_.querySelectorAll(strings.ICON_SELECTOR), (icon: HTMLElement) => {
  const ripple = MDCRipple.attachTo(icon);
  ripple.unbounded = true;
  return ripple;
}) as MDCRipple[];
```

If we return `"foo"` here instead of `ripple`, TypeScript won't yell at us 🙁